### PR TITLE
Change instance sort order to be more evenly distributed 

### PR DESCRIPTION
**Pull request by Security-Chief-Odo**
Also changed the weighting for 'active_monthly_users'. This should more effectively randomize the instance list, but also 'weight' or 'bias' the more popular instances.

In this way, more Lemmy instances will have more 'air-time' on the join-lemmy site instead of the same constant ones.

### CODE REVIEW

**Comment by Security-Chief-Odo** [comment]
Hey all, don't know why woodpecker is failing on the prettier and yaml linting here. I ran both commands with write multiple times on my local branch with no issues reported. 

```
Checking formatting...
All matched files use Prettier code style!
```

File: crawl.mjs [hunk 1]
```diff
@@ -4,14 +4,14 @@ import { spawn } from "child_process";
 const outDir = "src/shared/translations/";
 const recommendationsFile = "recommended-instances.json";
 const instanceStatsFile = "src/shared/instance_stats.ts";
-const min_monthly_users = 5;
+const min_monthly_users = 10;
```
Comment (Nutomic) [inline crawl.mjs:7]: Revert this change.

File: src/shared/components/instances.tsx [hunk 2]
```diff
@@ -27,22 +30,35 @@ export class Instances extends Component<any, any> {
...
+
     for (var i of instance_stats.stats.instance_details) {
       if (recommended_instances.indexOf(i.domain) > -1) {
         recommended.push(i);
       } else {
         remaining.push(i);
       }
+
+      // Min number of active users is 10 to include in this calculation
+      if (i.site_info.site_view.counts.users_active_month > 10) {
```
Comment (Nutomic) [inline src/shared/components/instances.tsx:43]: Use the constant for min active users.

File: src/shared/components/instances.tsx [hunk 3]
```diff
@@ -27,22 +30,35 @@ export class Instances extends Component<any, any> {
...
+        values.push(i.site_info.site_view.counts.users_active_month);
+      }
```
Comment (Nutomic) [inline src/shared/components/instances.tsx:45]: Instances with less active users are already filtered out by the crawl script, no need to check it here.

File: src/shared/components/instances.tsx [hunk 4]
```diff
@@ -27,22 +30,35 @@ export class Instances extends Component<any, any> {
...
     }
-    // shuffle recommended instances list into random order
-    // https://stackoverflow.com/a/46545530
+
+    // Use these values for the shuffle
+    const max = Math.max(...values);
+    const min = Math.min(...values);
+    const avg = this.averageFunc(values);
```
Comment (Nutomic) [inline src/shared/components/instances.tsx:51]: Ah so this is calculating `averageMonthlyUsers`, `maxMonthlyUsers` and `minMonthlyUsers` across all instances. Can you rename the vars to clarify this?

**Comment by Nutomic** [comment]
The new ordering looks great, thanks! Just needs a couple of code changes and fixing conflicts with main branch.

For formatting make sure to run prettier 3.0

File: crawl.mjs (hunk 1 above)
Comment (Security-Chief-Odo) [reply crawl.mjs:7]: Done, that was left over from testing, apologies.

File: src/shared/components/instances.tsx (hunk 4 above)
Comment (Security-Chief-Odo) [reply src/shared/components/instances.tsx:51]: Renamed.

**Comment by Nutomic** [comment]
The formatting is failing on `generate_translations.mjs`:
```
@@ -33,7 +33,7 @@ fs.readdir(translationDir, (_err, files) => {
const lang = filename.split(".")[0];
try {
const json = JSON.parse(
-        fs.readFileSync(translationDir + filename, "utf8")
+        fs.readFileSync(translationDir + filename, "utf8"),
);
let data = `export const ${lang} = {\n  translation: {`;
for (const key in json) {
```

**Comment by Nutomic** [comment]
Thanks!

**Comment by Nutomic** [comment]
The random factor on this looks way too high, Im seeing instances with 7 monthly users at the very top of the list. How can we adjust that?
